### PR TITLE
RND203-308: Add mutex lock in athn_reset and enable part of athn_attach

### PR DESCRIPTION
**Pull request by Mikolaj-Filar-Mobica**

### CODE REVIEW

**Review by Rafal-Litwinski-Mobica — APPROVED** [review]

File: sys/dev/athn/athn.c [hunk 1]
```diff
@@ -243,17 +243,21 @@ athn_attach(struct athn_softc *sc)
...
-		// TOTO missing field 'dv_xname' in 'struct device'
-		// printf("%s: could not reset chip\n", sc->sc_dev.dv_xname);
+		printf("Could not reset chip\n");
 		return (error);
 	}
 
 	if ((error = athn_set_power_awake(sc)) != 0) {
-		// TOTO missing field 'dv_xname' in 'struct device'
-		// printf("%s: could not wakeup chip\n", sc->sc_dev.dv_xname);
+		printf("Could not wakeup chip\n");
```
Comment (nzulinski-mobica) [inline sys/dev/athn/athn.c]: I think there should be dev_printf

File: sys/dev/athn/athn.c [hunk 2]
```diff
@@ -735,6 +740,7 @@ athn_reset(struct athn_softc *sc, int cold)
 	}
```
Comment (nzulinski-mobica) [inline sys/dev/athn/athn.c:741]: Also on line 738 there should be ATHN_UNLOCK()

File: sys/dev/athn/athn.c (hunk 1 above)
Comment (Mikolaj-Filar-Mobica) [reply sys/dev/athn/athn.c]: Done, thx!

File: sys/dev/athn/athn.c (hunk 2 above)
Comment (Mikolaj-Filar-Mobica) [reply sys/dev/athn/athn.c:741]: Done, thx!

**Pull request merged**